keepsorted/options_test.go

```diff
@@ -87,6 +87,24 @@ func TestBlockOptions(t *testing.T) {
 				PrefixOrder: []string{"a", "b", "c", "d"},
 			},
 		},
+		{
+			name:           "ItemList_YAML",
+			in:             `prefix_order=[a, b, c d, 'e",\f', 'g]', "h\"]"]`,
```

Comment:
Suggestion: I can think of a few ways to make this even more complete.

- Instead of `'e",\f'`, how about `'e",\'`, which better tests that `\'` does _not_ escape single quotes.
- Include a `\\` in the double quoted string
- Include an opening `[` in a quoted string. Maybe 2 of them, so that if future code ever incorrectly parsed them, the depth would still be uneven.

Reply:
I added a bunch of tests to the nest options_parser_test.go. PTAL